src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -140,37 +145,63 @@ export interface IMatchCirclesProps {
 }
 
 export function matchCirclesToData(props: IMatchCirclesProps) {
+  const { dataConfiguration, enableAnimation, instanceId, dotsElement } = props;
+  const allCaseData = dataConfiguration.joinedCaseDataArrays;
 
-  const {dataConfiguration, enableAnimation, instanceId,
-      dotsElement, pointRadius, pointColor, pointStrokeColor} = props,
-    allCaseData = dataConfiguration.joinedCaseDataArrays,
-    caseDataKeyFunc = (d: CaseData) => `${d.plotNum}-${d.caseID}`,
-    circles = selectCircles(dotsElement);
-  if (!circles) return;
+  // Create the circles
+  const allCircles = selectGraphDots(dotsElement);
+  if (!allCircles) return;
   startAnimation(enableAnimation);
-  circles
-    .data(allCaseData, caseDataKeyFunc)
+
+  allCircles
+    .data(allCaseData)
     .join(
-      (enter) =>
-        enter.append('circle')
+      enter => {
+        const g = enter.append('g')
           .attr('class', 'graph-dot')
-          .property('id', (d: CaseData) => `${instanceId}_${d.caseID}`),
-      (update) =>
-        update.attr('r', pointRadius)
-          .style('fill', pointColor)
-          .style('stroke', pointStrokeColor)
-          .style('stroke-width', defaultStrokeWidth)
-    );
-  dotsElement && select(dotsElement).on('click',
-    (event: MouseEvent) => {
-      const target = select(event.target as SVGSVGElement);
-      if (target.node()?.nodeName === 'circle') {
-        handleClickOnDot(event, (target.datum() as CaseData).caseID, dataConfiguration.dataset);
+          .property('id', (d: CaseData) => `${instanceId}_${d.caseID}`);
+        g.append('circle')
+          .attr('class', 'outer-circle');
+        g.append('circle')
+          .attr('class', 'inner-circle');
+        return g;
       }
```

Comment:
Looks good -- I'm a little surprised we don't need an update function, but I guess that's handled elsewhere.